.github/workflows/Pipeline.yml

```diff
@@ -1,450 +1,116 @@
     needs:
+      - Params
       - VerifyDocs
-
-    env:
-      ARTIFACT: pyEDAA-ProjectModel-documentation
-    outputs:
-      artifact: ${{ env.ARTIFACT }}
-
-    steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
-
-      - name: 🚢 Build container image 'vhdl/doc'
-        run: |
-          docker build -t edaa/doc - <<-EOF
-          FROM btdi/sphinx:featured
-          RUN apk add -U --no-cache graphviz
-          EOF
-
-      - name: 🛳️ Build documentation using container edaa/doc
-        uses: buildthedocs/btd@v0
-        with:
-          skip-deploy: true
-
-      - name: 📤 Upload 'documentation' artifacts
-        uses: actions/upload-artifact@master
-        with:
-          name: ${{ env.ARTIFACT }}
-          path: doc/_build/html
-          retention-days: 7
+    with:
+      artifact: ${{ fromJson(needs.Params.outputs.params).artifacts.doc }}
 
   PublishToGitHubPages:
-    name: 📚 Publish to GH-Pages
-    runs-on: ubuntu-latest
+    uses: pyTooling/Actions/.github/workflows/PublishToGitHubPages.yml@r0
     needs:
+      - Params
       - BuildTheDocs
       - Coverage
       - StaticTypeCheck
-
-    env:
-      DOC:      ${{ needs.BuildTheDocs.outputs.artifact }}
-      COVERAGE: ${{ needs.Coverage.outputs.artifact }}
-      TYPING:   ${{ needs.StaticTypeCheck.outputs.artifact }}
-    outputs:
-      artifact: ${{ env.ARTIFACT }}
-
-    steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
-
-      - name: 📥 Download artifacts '${{ env.DOC }}' from 'BuildTheDocs' job
-        uses: actions/download-artifact@v2
-        with:
-          name: ${{ env.DOC }}
-          path: public
-
-      - name: 📥 Download artifacts '${{ env.COVERAGE }}' from 'Coverage' job
-        uses: actions/download-artifact@v2
-        with:
-          name: ${{ env.COVERAGE }}
-          path: public/coverage
-
-      - name: 📥 Download artifacts '${{ env.TYPING }}' from 'StaticTypeCheck' job
-        uses: actions/download-artifact@v2
-        with:
-          name: ${{ env.TYPING }}
-          path: public/typing
-
-      - name: '📓 Publish site to GitHub Pages'
-        if: github.event_name != 'pull_request'
-        run: |
-          cd public
-          touch .nojekyll
-          git init
-          cp ../.git/config ./.git/config
-          git add .
-          git config --local user.email "BuildTheDocs@GitHubActions"
-          git config --local user.name "GitHub Actions"
-          git commit -a -m "update ${{ github.sha }}"
-          git push -u origin +HEAD:gh-pages
+    with:
+      doc: ${{ fromJson(needs.Params.outputs.params).artifacts.doc }}
+      coverage: ${{ fromJson(needs.Params.outputs.params).artifacts.coverage }}
+      typing: ${{ fromJson(needs.Params.outputs.params).artifacts.typing }}
 
   ArtifactCleanUp:
-    name: 🗑️ Artifact Cleanup
-    runs-on: ubuntu-latest
+    uses: pyTooling/Actions/.github/workflows/ArtifactCleanUp.yml@r0
     needs:
+      - Params
```

Comment:
`UnitTesting` as an artifact producer is missing, right?

What about the publishing of test results job which is a end-job?
-> Is this job missing?

See: https://github.com/pyTooling/pyTooling/blob/main/.github/workflows/Pipeline.yml#L397-L425

Reply:
I added `UnitTesting` to the `needs` of `ArtifactCleanUp`. Good catch!

With regard to publishing the test results, that was left out of these changes. When I started working on the reusable workflows a week ago, the "publishing test results was inconsistent:

- Not done (here): https://github.com/edaa-org/pyEDAA.ProjectModel/blob/main/.github/workflows/Pipeline.yml
- Done in job UnitTesting using EnricoMi/publish-unit-test-result-action (in pySVModel and pyVHDLModel): https://github.com/edaa-org/pySVModel/blob/main/.github/workflows/Pipeline.yml and https://github.com/VHDL/pyVHDLModel/blob/main/.github/workflows/Pipeline.yml
- Done in a separated job using dorny/test-reporter (in pyTooling): https://github.com/pyTooling/pyTooling/blob/main/.github/workflows/Pipeline.yml#L397-L425

Hence, I decided to delay it after we have all the common jobs done.

Reply:
Discussion was moved here: https://github.com/edaa-org/pyEDAA.ProjectModel/issues/36